relm4-macros/src/attrs.rs

```diff
@@ -93,28 +120,65 @@ impl Parse for Attrs {
 
         let mut attrs = Attrs::new();
 
-        // Attrs::stream_peek(input.cursor());
-
-        if input.peek(Token![pub]) {
-            let pub_vis = if input.is_empty() {
-                None
-            } else {
-                Some(input.parse()?)
-            };
-            attrs.visibility.pub_vis = pub_vis;
-            attrs.visibility_set = true;
-        }
-        else if input.peek(Ident) && input.peek2(Token![=]){
-            let _ident: Ident = input.parse()?;
-            let _eq: Token![=] = input.parse()?;
-
-            let pub_vis = if input.is_empty() {
-                None
+        while !input.is_empty() {
+
+            // Attrs::stream_peek(input.cursor());
+
+            if input.peek(Token![pub]) {
+                if attrs.visibility_set {
+                    return Err(input.error("You can't assign visibility twice"));
+                }
+                let pub_vis = if input.is_empty() {
+                    None
+                } else {
+                    Some(input.parse()?)
+                };
+                attrs.visibility = pub_vis;
+                attrs.visibility_set = true;
+            }
+            else if input.peek(Ident) && input.peek2(Token![=]){
+                // This allows to report errors at the argument name instead of value
+                let visibility_error = Err(input.error("You can't assign visibility twice"));
+                let relm4_error = Err(input.error("You can't assign relm4 path twice"));
+                let unknown_identifier_error = Err(input.error("Unknown argument. Valid arguments are: `visibility`, `relm4` and `gtk` "));
+
+                let ident: Ident = input.parse()?;
+                let _eq: Token![=] = input.parse()?;
+
+                let ident_value: String = ident.to_string();
```

Comment:
You can compare `Ident` with `&str` even without converting it to a sting before AFAIK.